watertap/unit_models/MD/membrane_distillation_base.py

```diff
@@ -358,11 +571,44 @@ def eq_flux_hot_enth(b, t, x):
             doc="cold side evaporation enthalpy flux",
         )
         def eq_flux_cold_enth(b, t, x):
-            return (
-                b.flux_enth_cold[t, x]
-                == b.flux_mass[t, x]
-                * b.cold_ch.properties_vapor[t, x].enth_mass_phase["Vap"]
-            )
+            if self.config.MD_configuration_Type == MDconfigurationType.DCMD:
+                return (
+                    b.flux_enth_cold[t, x]
+                    == b.flux_mass[t, x]
+                    * b.cold_ch.properties_vapor[t, x].enth_mass_phase["Vap"]
+                )
+            elif self.config.MD_configuration_Type == MDconfigurationType.PGMD_CGMD:
+                return (
+                    b.flux_enth_cold[t, x]
+                    == b.flux_mass[t, x]
+                    * b.gap_ch.properties[t, x].enth_mass_phase["Liq"]
+                )
+            elif self.config.MD_configuration_Type == MDconfigurationType.VMD:
+                return (
+                    b.flux_enth_cold[t, x]
+                    == b.flux_mass[t, x]
+                    * b.cold_ch.properties[t, x].enth_mass_phase["Vap"]
+                )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.difference_elements,
+            doc="vapor expansion heat flux",
+        )
+        def eq_flux_expansion_heat(b, t, x):
+            if self.config.MD_configuration_Type == MDconfigurationType.VMD:
+
+                R = 8.3146261 * pyunits.J / pyunits.mol / pyunits.K
```

Comment:
`R` can just be imported `from idaes.core.util.constants import Constants` (but not a dealbreaker)
```suggestion
                Constants.gas_constant
```